javaparser-core/src/main/java/com/github/javaparser/utils/SourceRoot.java

```diff
@@ -50,6 +46,9 @@
     private JavaParser javaParser = new JavaParser();
 
     public SourceRoot(Path root) {
+        if (Files.isRegularFile(root)) {
```

Comment:
another straightforward way of saying this would be `if (!Files.isDirectory(root))` but I guess this is just a minor comment. Overall, looks good to me.